app/javascript/hooks/mutations/room_settings/useUpdateRoomSetting.jsx

```diff
@@ -2,7 +2,15 @@ import { useMutation, useQueryClient } from 'react-query';
 import axios from '../../../helpers/Axios';
 
 export default function useUpdateRoomSetting(friendlyId) {
-  const patchRoomSetting = (roomSettingData) => axios.patch(`/room_settings/${friendlyId}.json`, roomSettingData);
+  const patchRoomSetting = (roomSettingData) => {
+    // If guestPolicy setting is toggled, bool are rewritten to string values as per BBB API.
+    if (roomSettingData.settingName === 'guestPolicy') {
+      Object.defineProperty(roomSettingData, 'settingValue', {
+        value: roomSettingData.settingValue === true ? 'ASK_MODERATOR' : 'ALWAYS_ACCEPT',
```

Comment:
What does `Object.defineProperty` do? Why's it needed here?

Reply:
esf will complain about "re-assigning" a var name if I try to overwrite a var.
This is in progress as I think I am over complicating things.